### PR TITLE
RA-740 "Automating RA-714 Review patient to avoid creating duplicates…

### PR DESCRIPTION
**Pull request by tomaszmueller**

### CODE REVIEW

**Comment by tomaszmueller** [comment]
@rkorytkowski https://issues.openmrs.org/browse/RA-740

File: ui-tests/src/test/java/org/openmrs/reference/DuplicatePatientRegisterTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        patient.address1 = "address";
+        patient.address2 = "address";
+        registrationPage.enterPatient(patient);
+    }
+    @After
+    public void tearDown() throws Exception {
+        if(registeredPatientId != null) {
+            deletePatient(registeredPatientId);
+        }
+        Thread.sleep(5000);	// a bit of a hack, wait for "created patient" popup to disappear
```
Comment (rkorytkowski) [inline ui-tests/src/test/java/org/openmrs/reference/DuplicatePatientRegisterTest.java]: It's unacceptable. You could implement a waiter in https://github.com/openmrs/openmrs-contrib-uitestframework/blob/master/src/main/java/org/openmrs/uitestframework/page/AbstractBasePage.java e.g. waitForHidingElementById

Comment (djazayeri) [inline ui-tests/src/test/java/org/openmrs/reference/DuplicatePatientRegisterTest.java]: A random example of waiting in another test:
https://github.com/openmrs/openmrs-distro-referenceapplication/blob/master/ui-tests/src/main/java/org/openmrs/reference/page/PatientDashboardPage.java#L26

**Comment by tomaszmueller** [comment]
@rkorytkowski 
I added waitForHidingElementBy method at
https://github.com/openmrs/openmrs-contrib-uitestframework/pull/1
to use it here to wait for "created patient" popup to disappear

**Comment by tomaszmueller** [comment]
@rkorytkowski Updated CoreApps and UITestFramework versions

File: ui-tests/src/test/java/org/openmrs/reference/DuplicatePatientRegisterTest.java (hunk 1 above)
Comment (rkorytkowski) [inline ui-tests/src/test/java/org/openmrs/reference/DuplicatePatientRegisterTest.java]: @tomaszmueller, please use the waitForElementToBeHidden method you added to uitestframework here.

Comment (tomaszmueller) [inline ui-tests/src/test/java/org/openmrs/reference/DuplicatePatientRegisterTest.java]: @rkorytkowski done

**Comment by tomaszmueller** [comment]
@rkorytkowski usage of waitForElementToBeHidden is added

**Pull request merged**